modules/meshrenderinggl/glsl/oit/display.frag

```diff
@@ -80,24 +87,29 @@ void main() {
         PickingData = vec4(0.0, 0.0, 0.0, 1.0);
         gl_FragDepth = p.depth;
 #else
-        // Copy fragments in local array
-        int numFrag = 0;
-        fillFragmentArray(pixelIdx, numFrag);
-        // Sort fragments in local memory array
-        bubbleSort(numFrag);
+        float backgroundDepth = 1.0;
+#ifdef BACKGROUND_AVAILABLE
+        // Assume the camera used to render the background has the same near and far plane,
+        // so we can directly compare depths.
+        vec2 texCoord = (gl_FragCoord.xy + 0.5) * reciprocalDimensions;
+        backgroundDepth = texture(bgDepth, texCoord).x;
+#endif  // BACKGROUND_AVAILABLE
 
         // front-to-back shading
         vec4 color = vec4(0);
-        for (int i = 0; i < numFrag; ++i) {
-            vec4 c = uncompressPixelData(fragmentList[i]).color;
+        vec4 nextFragment = selectionSortNext(pixelIdx, 0.0);
+        abufferPixel unpackedFragment = uncompressPixelData(nextFragment);
+        gl_FragDepth = min(backgroundDepth, unpackedFragment.depth);
+
+        while (unpackedFragment.depth >= 0 && unpackedFragment.depth <= backgroundDepth) {
+            vec4 c = unpackedFragment.color;
             color.rgb = color.rgb + (1 - color.a) * c.a * c.rgb;
             color.a = color.a + (1 - color.a) * c.a;
+
+            nextFragment = selectionSortNext(pixelIdx, unpackedFragment.depth);
+            unpackedFragment = uncompressPixelData(nextFragment);
         }
-        float depth = 1.0;
-        if (numFrag > 0) {
-            depth = uncompressPixelData(fragmentList[0]).depth;
-        }
-        gl_FragDepth = depth;
+
         FragData0 = color;
```

Comment:
Blending is set to alpha, so the fragment ops take care of that for us. Since we got the color texture bound as a side effect of binding depth, we could of course blend here ourselves, but I don't see an advantage of that, it's just more code.